MIPS/mip-x.md

```diff
@@ -0,0 +1,246 @@
+and separated into two steps.
+
+The first is for the Liquidator to sell a proportion of the amount it holds
+when the `triggerLiquidation` function is called. The `triggerLiquidation` is a
+public function and may be called by anyone, but may only be called once in a
+24 hour period. Based on the spot price for the asset the function will compute
+a value between $1000 and $3000 and trigger a sale for this amount. After the
+token has been swapped the token remains on the Liquidator contract.
+
+The second step is for the Integration contract to call `collect()` on the
+liquidator. This function will transfer a proportion of the tokens from the
+Liquidator contract to the Integration contract. At this point the token is on
+the Integration contract and earning Supply APY.
+
+The following psueudo code is a proposal for how `collect()` function should
+distribute bought tokens to the Integration contract when called. 
+
+```
+contractBal = buyAsset.balanceOf(address(this)) 
+spotPrice = exchange.getSpotPriceFor(buyAsset)
+value = contactBal * spotPrice
+// If the value is less than 1000 transfer everything
+if value < 1000
+  buyAsset.transfer(contractBal, integration)
+// If the value is gt 1000 transfer a percentage of the tokens as set by governance
+else
+  amountToSend = contractBal * collectPercentage
+  buyAsset.transfer(amountToSend, integration)
+```
+
+It is proposed that the `collect()` function is called at a specific time
+interval. The initial proposal is one hour. 
+  
+
+**Interface**
+
+```
+pragma solidity 0.5.16;
+
+interface ILiquidations {
+    struct Liquidation {
+        address basset;
+        address integration;
+        address rewardToken;
+        uint    amount;
+        uint    collectDrip;
+        bool    paused;
+    }
+
+    // Views
+    function getLiquidation(address _bAsset) external view returns (Liquidation memory l);
+
+    // Restricted to the `integrationContract` given in addLiquidation
+    function collect() external;
+
+    // Callable by anyone to trigger a selling event 
+    function triggerLiquidation(address _bAsset) public;
+
+    // Governor only
+    function addLiquidation(address _bAsset, address _integration, uint _amount ) external;
+    function removeLiquidation(address _bAsset) external;
+    function pauseLiquidation(address _bAsset) external;
+    function setCollectDrip(address _bAsset) external;
+}
+```
+**Events**
+
+* event LiquidationAdded(address indexed bAsset);
+* event LiquidationRemoved(address indexed bAsset);
+* event LiquidationPaused(address indexed bAsset);
+* event LiquidationTriggered(address indexed bAsset);
+* event LiquidationCollected(address indexed bAsset);
+
+### Compound Integration Contract
+---
+The `CompoundIntegration.sol` will be amend the `deposit` and `withdrawal` function to call `collect(_bAsset)` on the Liquidator contract. No interface changes are needed. 
+
+### Test Cases
```

Comment:
Probs not necessary to write these test cases given they can just be included more verbosely directly in the implementation